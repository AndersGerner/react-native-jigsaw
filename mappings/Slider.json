{"name":"Slider","tag":"Slider","description":"A component used to set a value in a range","category":"input","preview_image_url":"https://res.cloudinary.com/altos/image/upload/draftbit/library/jigsaw-1.0/reps/Control_Slider.png","supports_list_render":false,"props":{"disabled":{"label":"Disabled","description":"Whether the slider is disabled","editable":true,"required":false,"type":"boolean","value":null},"maximumValue":{"label":"Maximum value","description":"The maximum value of the slider","editable":true,"required":false,"type":"number","min":0,"max":1000000,"step":1,"precision":0,"value":10},"minimumValue":{"label":"Minimum value","description":"The minimum value of the slider","editable":true,"required":false,"type":"number","min":-1000000,"max":1000000,"step":1,"precision":0,"value":0},"step":{"label":"Step","description":"The amount the value should change per step","editable":true,"required":false,"type":"number","min":0,"max":100,"step":0.01,"precision":2,"value":null},"fieldName":{"label":"Field name","description":"The name of the field within the screen that will store this component's value","type":"fieldName","value":"sliderValue","valuePropName":"value","handlerPropName":"onValueChange","editable":true,"required":false},"maximumTrackTintColor":{"label":"Maximum Color","description":"Color of the track from the right of the thumb","editable":true,"required":false,"type":"color","value":null},"minimumTrackTintColor":{"label":"Minimum Color","description":"Color of the track from the left of the thumb","editable":true,"required":false,"type":"color","value":null},"thumbTintColor":{"label":"Thumb Tint Color","description":"Color of the thumb","editable":true,"required":false,"type":"color","value":null},"disabledThumbTintColor":{"label":"Disabled Thumb Tint Color","description":"Color of the thumb when the slider is disabled","editable":true,"required":false,"type":"color","value":null},"thumbTouchSize":{"label":"Hit Slop","description":"The size of the hit slop","editable":true,"required":false,"type":"number","min":0,"max":100,"step":1,"precision":2,"value":null},"trackBorderRadius":{"label":"Track Border Radius","description":"The border radius for the track ","editable":true,"required":false,"type":"number","min":0,"max":50,"step":1,"precision":0,"value":10},"thumbBorderRadius":{"label":"Thumb Border Radius","description":"The border radius for the thumb","editable":true,"required":false,"type":"number","min":0,"max":50,"step":1,"precision":0,"value":10},"thumbSize":{"label":"Thumb Size","description":"Size of the thumb","editable":true,"required":false,"type":"number","min":0,"max":100,"step":1,"precision":2,"value":null}},"layout":{}}