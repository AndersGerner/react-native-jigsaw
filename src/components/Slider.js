/* @flow */

import * as React from "react"
import Slider from "react-native-slider"
import { withTheme } from "../core/theming"
import type { Theme } from "../types"
import { FORM_TYPES, COMPONENT_TYPES, FIELD_NAME } from "../core/component-types"

type Props = {
  disabled?: boolean,
  maximumValue?: number,
  minimumValue?: number,
  step?: number,
  onValueChange: () => void,
  value: number,
  style?: any,
  /**
   * @optional
   */
  theme: Theme
}

class Slider extends React.Component {
  render() {
    const {
      style, //
      value,
      minimumTrackTintColor, //
      maximumTrackTintColor, //
      thumbTintColor, //
      minimumValue, //
      maximumValue, //
      thumbTouchSize, //
      step, //
      disabled, //
      onValueChange, //
      theme,
      ...props
    } = this.props
    const { colors } = theme

    return (
      <Slider
        disabled={disabled}
        value={value}
        minimumTrackTintColor={colors.primary}
        maximumTrackTintColor={maximumTrackTintColor}
        thumbTintColor={thumbTintColor}
        thumbTouchSize={{ width: thumbTouchSize, height: thumbTouchSize }}
        onValueChange={onValueChange}
        step={step}
        {...props}
        style={style}
      />
    )
  }
}

export default withTheme(Slider)

export const SEED_DATA = {
  name: "Slider",
  tag: "Slider",
  description: "A component used to set a value in a range",
  category: COMPONENT_TYPES.formControl,
  preview_image_url: "{CLOUDINARY_URL}/Control_Slider.png",
  supports_list_render: false,
  props: {
    disabled: {
      label: "Disabled",
      description: "Whether the slider is disabled",
      editable: true,
      required: false,
      type: FORM_TYPES.boolean,
      value: null
    },
    maximumValue: {
      label: "Maximum value",
      description: "The maximum value of the slider",
      editable: true,
      required: false,
      type: FORM_TYPES.number,
      min: 0,
      max: 1000000,
      step: 1,
      precision: 0,
      value: 10
    },
    minimumValue: {
      label: "Minimum value",
      description: "The minimum value of the slider",
      editable: true,
      required: false,
      type: FORM_TYPES.number,
      min: -1000000,
      max: 1000000,
      step: 1,
      precision: 0,
      value: 0
    },
    step: {
      label: "Step",
      description: "The amount the value should change per step",
      editable: true,
      required: false,
      type: FORM_TYPES.number,
      min: 0,
      max: 100,
      step: 0.01,
      precision: 2,
      value: null
    },
    fieldName: {
      ...FIELD_NAME,
      value: "sliderValue",
      handlerPropName: "onValueChange"
    },
    maximumTrackTintColor: {
      label: "Maximum Color",
      description: "Color of the track from the right of the thumb",
      editable: true,
      required: false,
      type: FORM_TYPES.color,
      value: null
    },
    minimumTrackTintColor: {
      label: "Minimum Color",
      description: "Color of the track from the left of the thumb",
      editable: true,
      required: false,
      type: FORM_TYPES.color,
      value: null
    },
    thumbTintColor: {
      label: "Thumb Tint Color",
      description: "Color of the thumb",
      editable: true,
      required: false,
      type: FORM_TYPES.color,
      value: null
    },
    thumbTouchSize: {
      label: "Hit Slop",
      description: "The size of the hit slop",
      editable: true,
      required: false,
      type: FORM_TYPES.number,
      min: 0,
      max: 100,
      step: 0.01,
      precision: 2,
      value: null
    },
    onValueChange: {
      label: "Action",
      description: "Action to execute when button pressed",
      editable: true,
      type: FORM_TYPES.action,
      value: null
    }
  },
  layout: {
    width: 375,
    height: 24
  }
}
