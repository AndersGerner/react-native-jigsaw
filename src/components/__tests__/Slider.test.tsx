import * as React from "react";
import renderer from "react-test-renderer";
import Slider from "../Slider";

it("renders slider", () => {
  const tree = renderer
    .create(
      <Slider
        style={{ height: 4 }}
        maximumTrackTintColor="primary"
        minimumTrackTintColor="light"
        thumbTintColor="primary"
        minimumValue={0}
        maximumValue={10}
        step={1}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
