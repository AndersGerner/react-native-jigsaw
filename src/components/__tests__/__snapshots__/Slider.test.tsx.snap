// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders slider 1`] = `
<RNCSlider
  disabled={false}
  enabled={true}
  inverted={false}
  maximumTrackTintColor="rgba(90, 69, 255, 1)"
  maximumValue={10}
  minimumTrackTintColor="rgba(165, 173, 183, 1)"
  minimumValue={0}
  onChange={[Function]}
  onRNCSliderSlidingComplete={null}
  onRNCSliderSlidingStart={null}
  onRNCSliderValueChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  step={1}
  style={
    Array [
      Object {
        "height": 40,
      },
      Object {
        "height": 4,
      },
    ]
  }
  thumbImage={
    Object {
      "__packager_asset": true,
      "height": 1,
      "scale": 1,
      "uri": "file://assets/full/path/to/directory/name.type",
      "width": 1,
    }
  }
  thumbTintColor="rgba(90, 69, 255, 1)"
  value={0}
/>
`;
